### PR TITLE
Don't trigger provider scoped lints in tests

### PR DESCRIPTION
**Pull request by lockieRichter**
Updates the `scoped_providers_should_specify_dependencies` to ignore instances of using pumpWidget in tests

### CODE REVIEW

File: packages/riverpod_lint_flutter_test/pubspec.yaml [hunk 1]
```diff
@@ -21,7 +21,9 @@ dev_dependencies:
   riverpod_lint:
     path: ../riverpod_lint
   riverpod_generator: ^2.0.0
-  test: ^1.15.0
+  test: ^1.24.9
+  flutter_test:
```
Comment (rrousselGit) [inline packages/riverpod_lint_flutter_test/pubspec.yaml:25]: Why those changes and the ones in the pubspec_overrides?

Comment (rrousselGit) [reply packages/riverpod_lint_flutter_test/pubspec.yaml:25]: Wait ignore me, I get it

File: packages/riverpod_lint_flutter_test/pubspec_overrides.yaml [hunk 2]
```diff
@@ -14,3 +14,4 @@ dependency_overrides:
     path: ../riverpod_annotation
   riverpod_lint:
     path: ../riverpod_lint
+  test_api: ^0.6.1
```
Comment (rrousselGit) [inline packages/riverpod_lint_flutter_test/pubspec_overrides.yaml:17]: Why is this needed?

File: packages/riverpod_lint_flutter_test/test/lints/scoped_providers_should_specify_dependencies.dart [hunk 3]
```diff
@@ -121,6 +122,21 @@ void definitelyNotAMain() {
   );
 }
 
+void someTestFunction() {
+  testWidgets('override repositoryProvider in test', (tester) async {
+    await tester.pumpWidget(
+      ProviderScope(
```
Comment (rrousselGit) [inline packages/riverpod_lint_flutter_test/test/lints/scoped_providers_should_specify_dependencies.dart:128]: We need a test where `ProviderScope` is _not_ prevent in pumpWidget

And one where `ProviderScope` is prevent within pumpWidget, but is not the very first widget

File: packages/riverpod_lint/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,6 @@
+## Unreleased patch
+- Updated `scoped_providers_should_specify_dependencies` to ignore instances of using pumpWidget in tests
```
Comment (rrousselGit) [inline packages/riverpod_lint/CHANGELOG.md:2]:
```suggestion

- Updated `scoped_providers_should_specify_dependencies` to ignore instances of using pumpWidget in tests
```

**Review by rrousselGit — DISMISSED** [review]
LGTM once a few extra test cases are added

File: packages/riverpod_lint_flutter_test/pubspec_overrides.yaml (hunk 2 above)
Comment (lockieRichter) [reply packages/riverpod_lint_flutter_test/pubspec_overrides.yaml:17]: `flutter_test` has a pinned dependency on `test_api: 0.6.0` but the `test` package needs `test_api: ^0.61`. Needed to do this to get the packages all playing nicely.

File: packages/riverpod_lint_flutter_test/test/lints/scoped_providers_should_specify_dependencies.dart (hunk 3 above)
Comment (lockieRichter) [reply packages/riverpod_lint_flutter_test/test/lints/scoped_providers_should_specify_dependencies.dart:128]: Wil do, slick!

Comment (lockieRichter) [reply packages/riverpod_lint_flutter_test/test/lints/scoped_providers_should_specify_dependencies.dart:136]: Added some more tests covering these cases.

File: packages/riverpod_lint/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,7 @@
+## Unreleased patch
+
+- Updated `scoped_providers_should_specify_dependencies` to ignore instances of using pumpWidget in tests
```
Comment (rrousselGit) [inline packages/riverpod_lint/CHANGELOG.md:3]:
```suggestion
- Updated `scoped_providers_should_specify_dependencies` to ignore instances of using pumpWidget in tests (thanks to [lockieRichter](https://github.com/lockieRichter))
```

**Comment by rrousselGit** [comment]
Looking good, thanks!
